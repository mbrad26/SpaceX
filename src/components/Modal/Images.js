import React from 'react';

import { Image, Carousel } from 'react-bootstrap';

const Images = ({ activeItem }) => (
  <Carousel>
    {activeItem.flickr_images.map(url =>
      <Carousel.Item key={url}>
        <Image
          src={url}
          alt='First slide'
        />
      </Carousel.Item>
    )}
  </Carousel>
)

export default Images;
