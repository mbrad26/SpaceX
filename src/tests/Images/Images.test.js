import React from 'react';
import { render, screen } from '@testing-library/react';
import Images from '../../components/Modal/Images';
import { itemOne } from '../fixtures';

describe('Images', () => {
  it('renders snapshot', () => {
    const { container } = render(<Images activeItem={itemOne} />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders all the images', () => {
    render(<Images activeItem={itemOne} />);

    expect(screen.getAllByRole('img').length).toBe(2);
  });
});
