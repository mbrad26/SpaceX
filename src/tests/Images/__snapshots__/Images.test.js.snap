// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Images renders snapshot 1`] = `
<div
  class="carousel slide"
>
  <ol
    class="carousel-indicators"
  >
    <li
      class="active"
    />
    <li />
  </ol>
  <div
    class="carousel-inner"
  >
    <div
      class="active carousel-item"
    >
      <img
        alt="First slide"
        class=""
        src="url1"
      />
    </div>
    <div
      class="carousel-item"
    >
      <img
        alt="First slide"
        class=""
        src="url2"
      />
    </div>
  </div>
  <a
    class="carousel-control-prev"
    href="#"
    role="button"
  >
    <span
      aria-hidden="true"
      class="carousel-control-prev-icon"
    />
    <span
      class="sr-only"
    >
      Previous
    </span>
  </a>
  <a
    class="carousel-control-next"
    href="#"
    role="button"
  >
    <span
      aria-hidden="true"
      class="carousel-control-next-icon"
    />
    <span
      class="sr-only"
    >
      Next
    </span>
  </a>
</div>
`;
